openedx/core/djangoapps/user_api/accounts/serializers.py

```diff
@@ -63,6 +68,16 @@ def to_representation(self, user):
         :return: Dict serialized account
         """
         profile = user.profile
+        badges = settings.FEATURES.get('ENABLE_OPENBADGES') or False
+        if 'badges' in self.excluded_fields:
+            # Avoid doing this potentially expensive query-- it will be filtered out anyway.
```

Comment:
@nasthagiri I don't think it's likely to be all that expensive. It would make for larger JSON payloads, and it is possible that someone will have earned enough badges to cause a problem because it isn't paginated-- though unlikely, I'd think. It takes a while to earn a badge-- you can't just upload a bunch of them. There isn't a current plan to paginate in a follow-up PR.

However, I could remove this entirely from the User API, modify the Badges API to check if a user is sharing their profile fully, and display the badges if so. It and the subsequent PR won't be merged in time for Friday's release, however.